azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/network/DraftNetwork.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.compute.network;
+
+import com.azure.resourcemanager.network.models.Network.DefinitionStages.WithCreateAndSubnet;
+import com.microsoft.azure.toolkit.lib.common.model.Region;
+import com.microsoft.azure.toolkit.lib.compute.AzureResourceDraft;
+import com.microsoft.azure.toolkit.lib.compute.network.model.Subnet;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = true)
+public class DraftNetwork extends Network implements AzureResourceDraft<Network> {
+
+    private String subscriptionId;
+    private String resourceGroup;
+    private String name;
+    private Region region;
+    private String addressSpace;
+
+    private String subnet;
+    private String subnetAddressSpace;
+
+    public DraftNetwork(@Nonnull final String subscriptionId, @Nonnull final String resourceGroup, @Nonnull final String name) {
+        super(getResourceId(subscriptionId, resourceGroup, name), null);
```

Comment:
no id to create id for draft

Reply:
Let's redesign the draft pattern and apply to all resources later